### ISSUE TITLE
Inconsistent behavior when adding two "duplicate" functional terms with @formula

### ISSUE THREAD

**Issue by junyuan-chen** [JuliaStats/StatsModels.jl#204]
This seems to be a minor issue, but I decide to report it here anyway.
Here is an example when two functional terms that are supposed to be identical are handled differently with and without `@formula`.

```julia
julia> using StatsModels

julia> @formula (y ~ lag(x,1) + lag(x,1))
FormulaTerm
Response:
  y(unknown)
Predictors:
  (x)->lag(x, 1)
  (x)->lag(x, 1)

julia> @formula (y ~ x + x)
FormulaTerm
Response:
  y(unknown)
Predictors:
  x(unknown)

julia> term(:y) ~ lag(term(:x),1) + lag(term(:x),1)
FormulaTerm
Response:
  y(unknown)
Predictors:
  lag(x, 1)

julia> term(:y) ~ term(:x) + term(:x)
FormulaTerm
Response:
  y(unknown)
Predictors:
  x(unknown)
```

Since an instance of `FunctionalTerm` created by `@formula` involves an anonymous function, the two `lag(x,1)`s inside `@formula` are not considered identical and hence the "uniqueness" is not handled.

**Comment by kleinschmidt** [comment]
Thanks for the report, that's a real issue since we try to ensure that terms are
unique in a context, as you noted.  I also agree with your diagnosis of the
underlying issue, which (I think) should be fixed by #183.  In the mean time I
think we can add a specialized method for comparing two function terms...

**Comment by kleinschmidt** [comment]
This is fixed in v0.6.18 so `] up` should fix it https://github.com/JuliaRegistries/General/pull/27443
